modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/MongoDataStoreFactory.java

```diff
@@ -58,6 +58,13 @@
                     "Objects IDs for schema generation (comma separated)",
                     false,
                     null);
+    public static final Param DEACTIVATE_NATIVE_OR =
+            new Param(
+                    "deactivate_native_or",
+                    Boolean.class,
+                    "Deactivate native '$or' operator",
+                    false,
+                    Boolean.FALSE);
```

Comment:
This is a clean approach to backwards compatibility, thinking out loud, how will this play with migrations? Some upgrades from GS 2.14.x and has 5 MongoDB stores, by default $or will be activated right?

Reply:
Indeed native $or filter will be activated by default and allowing user for deactivating it on store configuration if needed.